File: 프로그래머스/2/42626. 더 맵게/더 맵게.py

import heapq as hq

def solution(scoville, k):
    answer = 0
    
    q = []
    
    for i in scoville:
        hq.heappush(q, i)
    
    count = 0
    while q:
        first = hq.heappop(q)
        
        if first >= k:
            return count
        
        if q:
            second = hq.heappop(q)
            new = first + (second * 2)
            count += 1
            hq.heappush(q, new)               
        else:
            break
                        
    return -1

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int k) {
        PriorityQueue<Integer> q = new PriorityQueue<>();
        for (int i = 0; i < scoville.length; i++) {
            q.add(scoville[i]);
        }
                
        int cnt = 0;
        while (!q.isEmpty()) {
            int first = q.poll();
                        
            if (first >= k) {
                return cnt;
            }
            
            if (!q.isEmpty()) {
                int second = q.poll();
                int newOne = first + (second * 2);
                cnt++;
                q.add(newOne);
            } else {
                break;
            }
        }
        
        return -1;
    }
}